src/cijoe/core/command.py

```diff
@@ -104,20 +108,27 @@ def _run(self, cmd, cwd, env, transport):
         os.makedirs(cmd_output_dpath, exist_ok=True)
 
         with open(cmd_output_fpath, "a", encoding=ENCODING) as logfile:
-            begin = time.time()
-            err = transport.run(cmd, cwd, env, logfile)
             state = CommandState(
                 cmd=cmd,
                 cwd=cwd,
-                err=err,
-                begin=begin,
-                end=time.time(),
+                err=0,
+                begin=0,
+                end=0,
                 output_dpath=cmd_output_dpath,
                 output_fpath=cmd_output_fpath,
                 state_fpath=cmd_state_fpath,
+                done=False,
             )
             state.to_file()
 
+            begin = time.time()
+            err = transport.run(cmd, cwd, env, logfile, self.monitor)
```

Comment:
Why is it required to move the creation of the **CommandState** before the execution?

Reply:
I need to call the `state.to_file()`, such that we step-data for the workflow report, if we generate it mid step (with `cijoe -p`)